### PR TITLE
Fix multiple build errors for NETCore on Windows.

### PR DESCRIPTION
**Pull request by lateralusX**

### CODE REVIEW

**Comment by EgorBo** [comment]
Btw, I think @filipnavara recently noticed that the windows lane is always green even if there are build errors

**Comment by filipnavara** [comment]
Changes in `netcore/System.Private.CoreLib/shared/` have to be submitted through CoreCLR, otherwise they won't be mirrored properly. As @EgorBo mentioned, the Azure Pipelines scripts lack error checking so the build is always green.

File: netcore/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems [hunk 1]
```diff
@@ -1076,7 +1076,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Kernel32\Interop.GetSystemTime.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Kernel32\Interop.GetSystemTimeAsFileTime.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Kernel32\Interop.GetSystemTimePreciseAsFileTime.cs" />
-    <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Kernel32\Interop.GetSystemTimes.cs" Condition="'$(FeaturePerfTracing)' == 'true'" />
+    <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Kernel32\Interop.GetSystemTimes.cs" Condition="'$(FeaturePerfTracing)' == 'true' AND '$(FeaturePortableThreadPool)' == 'false'" />
```
Comment (filipnavara) [inline netcore/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems]: Shouldn't this be 'OR' ?

File: netcore/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems [hunk 2]
```diff
@@ -1372,6 +1372,7 @@
     <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Kernel32\Interop.GetSystemTimes.cs" Condition="'$(TargetsWindows)' == 'true'" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Threading\LowLevelLifoSemaphore.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Threading\LowLevelLifoSemaphore.Windows.cs" Condition="'$(TargetsWindows)'=='true'" />
+    <Compile Include="$(MSBuildThisFileDirectory)Interop\Windows\Kernel32\Interop.CompletionPort.cs" Condition="'$(TargetsWindows)' == 'true'" />
```
Comment (filipnavara) [inline netcore/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems]: Thanks, I missed this while porting from CoreRT.

**Comment by filipnavara** [comment]
If you want me to submit this as CoreRT PR just ping me. I messed it up so I will be happy to fix it.

File: netcore/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems (hunk 1 above)
Comment (lateralusX) [reply netcore/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems]: So no FeaturePerfTracing with FeaturePortableThreadPool?

**Comment by lateralusX** [comment]
@filipnavara sure, you can do the CoreRT part and I can take a look how that is done.

File: netcore/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems (hunk 1 above)
Comment (filipnavara) [reply netcore/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems]: Oh, I misread it. :)

**Comment by filipnavara** [comment]
Submitted as https://github.com/dotnet/corert/pull/7849. Once it's approved and merged it a bot will open PR on Mono to mirror it.

**Comment by filipnavara** [comment]
Mirror PR: https://github.com/mono/mono/pull/17604

**Comment by lateralusX** [comment]
@filipnavara so now when the mirrored PR is merged, I guess what's left of this PR is the change outside the shared folder? So it is only changes in the shared folder needs to go through the CoreCLR/CoreRT workflow?

**Comment by filipnavara** [comment]
> so now when the mirrored PR is merged, I guess what's left of this PR is the change outside the shared folder?

Correct.

> So it is only changes in the shared folder needs to go through the CoreCLR/CoreRT workflow?

Yep. In this particular case it made sense to start from CoreRT since it needed local changes too and those could be tested in the CI of that repository.

I would have caught this earlier if the Windows builds were correctly reporting the status. Unfortunately I noticed it too late. I will look into fixing it.

**Comment by lateralusX** [comment]
Closed in favor of #17616

**Pull request closed without merging**